src/io/tree.cpp

```diff
@@ -416,12 +416,16 @@ std::string Tree::ToJSON() const {
   str_buf << "\"num_cat\":" << num_cat_ << "," << '\n';
   str_buf << "\"shrinkage\":" << shrinkage_ << "," << '\n';
   if (num_leaves_ == 1) {
+    str_buf << "\"tree_structure\":{";
     if (is_linear_) {
-      str_buf << "\"tree_structure\":{" << "\"leaf_value\":" << leaf_value_[0] << ", " << "\n";
-      str_buf << LinearModelToJSON(0) << "}" << "\n";
+      str_buf << "\"leaf_value\":" << leaf_value_[0] << ", " << '\n';
```

Comment:
It looks like the `leaf_value` line is now identical in both of these `if-else` branches. To simplify this a bit, could you move this up out of the `if-else`?